### PR TITLE
fix handling of long text columns

### PR DESCRIPTION
**Pull request by Anna050689**
fix handling of long text columns

### CODE REVIEW

File: src/syngen/ml/reporters/reporters.py [hunk 1]
```diff
@@ -76,16 +78,16 @@ def preprocess_data(self):
...
 
         int_columns = date_columns | int_columns
-
-        original = text_to_continuous(original, str_columns).drop(str_columns, axis=1)
-        synthetic = text_to_continuous(synthetic, str_columns).drop(str_columns, axis=1)
+        text_columns = str_columns | long_text_columns
+        original = text_to_continuous(original, text_columns).drop(text_columns, axis=1)
+        synthetic = text_to_continuous(synthetic, text_columns).drop(text_columns, axis=1)
 
         for col in [i + "_word_count" for i in str_columns]:
```
Comment (PaulBobyrev) [inline src/syngen/ml/reporters/reporters.py:85]: I guess it's better to replace [i + "_word_count" for i in **str_columns**] to [i + "_word_count" for i in **text_columns**]

Comment (Anna050689) [reply src/syngen/ml/reporters/reporters.py:85]: Thanks) I got it

File: src/syngen/ml/vae/models/dataset.py [hunk 2]
```diff
@@ -273,6 +275,28 @@ def _set_categorical_columns(self, df: pd.DataFrame, schema: Dict):
...
+            data_subset = df.select_dtypes(include=[pd.StringDtype(), "object"])
+        else:
+            text_columns = [
+                col for col, data_type in self.schema.get("fields", {}).items()
+                if data_type == "string"
+            ]
+            data_subset = df[text_columns]
+        self.long_text_columns = set()
+        if not data_subset.empty:
+            data_subset = data_subset.loc[:, data_subset.apply(lambda x: (x.str.len() > 200).any())]
```
Comment (PaulBobyrev) [inline src/syngen/ml/vae/models/dataset.py:292]: Have you tested the case when text columns exist but none of them is a long text feature? I have a concern that this data_subset.apply(lambda x: (x.str.len() > 200).any()) will be an empty list an will raise the "Cannot index with multidimensional index" error

**Review by PaulBobyrev — COMMENTED** [review]
Anya, please look at the comments

File: src/syngen/ml/vae/models/dataset.py (hunk 2 above)
Comment (Anna050689) [reply src/syngen/ml/vae/models/dataset.py:292]: I tested this case. It works okay)

**Pull request merged**